lib/autoupdate.ps1

```diff
@@ -349,85 +403,156 @@ function get_version_substitutions([String] $version, [Hashtable] $customMatches
     return $versionVariables
 }
 
-function autoupdate([String] $app, $dir, $json, [String] $version, [Hashtable] $matches) {
-    Write-Host -f DarkCyan "Autoupdating $app"
-    $has_changes = $false
-    $has_errors = $false
-    [Bool]$valid = $true
-    $substitutions = get_version_substitutions $version $matches
-
-    if ($json.url) {
-        # create new url
-        $url   = substitute $json.autoupdate.url $substitutions
-        $valid = $true
-
-        if($valid) {
-            # create hash
-            $hash = get_hash_for_app $app $json.autoupdate.hash $version $url $substitutions
-            if ($null -eq $hash) {
-                $valid = $false
-                Write-Host -f DarkRed "Could not find hash!"
-            }
-        }
+function Invoke-AutoUpdate {
+    param (
+        [String]
+        $App,
```

Comment:
What's the difference between $App and $AppName ? :confused:

Reply:
One thing... I change them all to `AppName`, more formal.